codice/ddf
# DDF-4570 Persist updated Solr password

#### What does this PR do?
This PR changes how the Solr password update module is invoked. The module is now invoked as part of the installation process. Prior, the module was invoked lazily when the system needed to create a SolrClient. The change is intended to prevent the installation process from overwriting the new Solr password when it saves changes to the custom.system.properties file. A side-effect of this PR is reduced dependencies in the SolrClient module, and modules that depend on it.

#### Who is reviewing it? 
@vinamartin @brjeter @austinsteffes @tyler30clemens 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter @brendan-hofmann 

#### How should this be tested?
1. Build DDF.
2. Go through the installation process and let DDF restart it self.

Expected behavior: DDF can communicate with Solr after the restart.
Post-condition: The property `solr.password` in file `etc/custom.system.properties` begins with the literal `ENC(`


Example:
```
aaron@aaron-dell:/sandbox/ddf-2.13.8-SNAPSHOT$ grep "solr.password" etc/custom.system.properties 
solr.password=admin
aaron@aaron-dell:/sandbox/ddf-2.13.8-SNAPSHOT$ grep "solr.password" etc/custom.system.properties 
solr.password = ENC(AWADVsbtqoPKVRVcDK5CsW2vSyvVTWbe4AzFQjbBycYx8KwEXk3KKyTQ7nONwWubqNFXyDniyUMkE9+05EZN3JM=)
```

#### Any background context you want to provide?
The PR https://github.com/codice/ddf/pull/4578 solved most reported instances of the issue. However, a few developers continue to report that the Solr password is successfully updated in Solr, but not persisted in the custom.system.properties file. It is likely that a timing issue or environmental peculiarity causes the read/write activities to the properties file happen out-of-order.
 
#### Checklist:
- [N/A] Documentation Updated
- [N/A] Update / Add Threat Dragon models
- [N/A] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
